runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1155,6 +1156,10 @@ public Instant getStateStartTime() {
     // Note that TimerOrElementCoder is a backwards-compatibility class
     // that is really a FakeKeyedWorkItemCoder
     Coder<?> valueCoder = ((WindowedValueCoder<?>) readCoder).getValueCoder();
+
+    if (valueCoder instanceof KvCoder<?, ?>) {
+      return ((KvCoder<?, ?>) valueCoder).getKeyCoder();
+    }
```

Comment:
I'm not sure I understand why this is needed.

Reply:
without it, the key passed into the `WindmillStateInternals` constructor is always null.  It doesn't look like it was ever really used before other than for debug messages.